### PR TITLE
5 - Edit Popup for Real Time Editing

### PR DESCRIPTION
**Pull request by laykevin**

### CODE REVIEW

File: dist.7z
Comment (joseph-w-valdez) [inline dist.7z:1]: delete this file

File: dist/manifest.json
Comment (joseph-w-valdez) [inline dist/manifest.json:1]: update the version to `1.0.0`

File: package.json
Comment (joseph-w-valdez) [inline package.json:1]: update the version to `1.0.0`

File: src/components/formButtons/FormButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                className={`${css.btnGrey} col-span-2`}
+                onClick={() => {
+                    setLiveSettings({ ...defaultSettings });
+                    sendMessageToTab("restoreSettings", {
+                        ...defaultSettings,
+                    });
+                    setIsEditing(true);
+                }}
+            >
+                Restore Default
```
Comment (joseph-w-valdez) [inline src/components/formButtons/FormButtons.tsx:39]: change to `Restore Defaults`

File: src/lib/index.ts
Comment (joseph-w-valdez) [inline src/lib/index.ts:1]: what is this file needed for?

File: src/shared/utils/stylingFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+let inputBoxMaxWidthStyle = "";
+let messageBoxColors = "";
+let messageColorUserStyle = "";
+let messageColorNonUserStyle = "";
+let textColorUserStyle = "";
+let textColorNonUserStyle = "";
+const selectionColors = "";
+let messageButtonsVisibilityStyle = ``;
+const codeSnippetWidth = `
+    [data-testid^="conversation-turn-"] > * > * > *:nth-child(2) {
```
Comment (joseph-w-valdez) [inline src/shared/utils/stylingFunctions.ts:15]: we should start creating variables to contain the strings pointing to different points on the page and using then using them for all of these styling functions

File: src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+          Object {
+            "backgroundColor": "rgba(145, 10, 103, 0.1)",
+          }
+        }
+        type="text"
+        value="50"
+      />
+      <input
+        className="absolute w-full h-0.5 left-0 bottom-0"
+        max="95"
```
Comment (joseph-w-valdez) [inline src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap:40]: the cap should be changed to `100`

File: src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+          Object {
+            "backgroundColor": "rgba(114, 4, 85, 0.1)",
+          }
+        }
+        type="text"
+        value="20"
+      />
+      <input
+        className="absolute w-full h-0.5 left-0 bottom-0"
+        max="95"
```
Comment (joseph-w-valdez) [inline src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap:97]: change this to `100`

File: src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+          Object {
+            "backgroundColor": "rgba(60, 7, 83, 0.1)",
+          }
+        }
+        type="text"
+        value="10"
+      />
+      <input
+        className="absolute w-full h-0.5 left-0 bottom-0"
+        max="95"
```
Comment (joseph-w-valdez) [inline src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap:154]: change this to `100`

File: src/popup/views/messageEditor/components/messageSliderControls/component.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        constructor(settingsProperties: InputSetting) {
+            this.name = settingsProperties.name;
+            this.id = settingsProperties.id;
+            this.valueType = settingsProperties.valueType;
+            this.bgColor = settingsProperties.bgColor;
+        }
+    }
+    const inputSettings = [
+        new InputSetting({
+            name: "Message Width",
```
Comment (joseph-w-valdez) [inline src/popup/views/messageEditor/components/messageSliderControls/component.tsx:33]: we should make the text unselectable so users can't highlight things like `Message Width`

File: src/popup/views/miscEditor/component.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    });
+
+    return (
+        <div className="flex flex-col h-full justify-between px-3 pb-2">
+            <div
+                className={`grid grid-cols-1 gap-y-3 w-full ${
+                    !isEditing ? "animate-fade-in" : ""
+                }`}
+            >
+                <label className="flex justify-between group p-2 hover:cursor-pointer hover:bg-slate-500/20 duration-100">
```
Comment (joseph-w-valdez) [inline src/popup/views/miscEditor/component.tsx:27]: let's add the tailwind class `select-none` so that users can't highlight the text `Show Chat Message Buttons`
also, let's add some styling to the background and border radius values so that users can tell that they can click on the entire button

**Review by joseph-w-valdez — CHANGES_REQUESTED** [review]

**Review by joseph-w-valdez — COMMENTED** [review]
LGTM - pre-emptively approving but adjust the values for the sliders to accept up to `100`. Make sure to squash. Resolve the comments when completed too.

**Review by joseph-w-valdez — APPROVED** [review]

File: src/shared/utils/stylingFunctions.ts [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+          ${messageBubbles} textarea {
+            padding: 3px;
+            background-color: rgba(0, 0, 0, 0.4);;
+            border-radius: 5px
+        }
+        `;
+    },
+    messageColorNonUserStyle: <T>(color: T) => {
+        messageColorNonUserStyle = `
+          ${messageBubbles}:nth-child(odd) > * > * { background-color: ${color} !important }
```
Comment (joseph-w-valdez) [inline src/shared/utils/stylingFunctions.ts:59]: we should also account for the `ul` and `ul code` elements here

File: src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap (hunk 3 above)
Comment (laykevin) [reply src/popup/views/messageEditor/components/messageSliderControls/__test__/__snapshots__/test_messageSliderControls.tsx.snap]: 👍

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## [1.0.0] - 2024-02-12
+
+### Added
+
```
Comment (laykevin) [inline CHANGELOG.md:6]: QA Lock

Comment (laykevin) [reply CHANGELOG.md:6]: REMINDER!! UPDATE CHANGELOG
```
Updated popup to be able to update in real time and save settings.
Popup saves settings when closed.
Refactored styling functions for scalability.
Updated textColorNonUserStyle function to include other types of text elements.
Updated max number values from 95 to 100.
```
